Foundation/NSJSONSerialization.swift

```diff
@@ -271,19 +281,24 @@ private struct JSONWriter {
     
     mutating func serializeJSON(_ obj: Any) throws {
 
+        // For better performance, the most expensive conditions to evaluate should be last.
         switch (obj) {
         case let str as String:
             try serializeString(str)
         case let boolValue as Bool:
             serializeBool(boolValue)
-        case _ where _SwiftValue.store(obj) is NSNumber:
-            try serializeNumber(_SwiftValue.store(obj) as! NSNumber)
+        case let num as Int:
+            writer(String(describing: num))
```

Comment:
This isn't as straightforward as I'd expected, and I need to give it some more thought.  In the meantime, I'll revert the fast paths from this PR as I'd like to deliver the 20% improvement from case reordering.